server/protocols/webadmin/webadmin-cassandra-data/src/test/java/org/apache/james/webadmin/routes/CassandraMappingsRoutesTest.java

```diff
@@ -0,0 +1,164 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.routes;
+
+import static io.restassured.RestAssured.given;
+import static io.restassured.RestAssured.when;
+import static io.restassured.RestAssured.with;
+import static org.apache.james.webadmin.WebAdminServer.NO_CONFIGURATION;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.apache.james.metrics.logger.DefaultMetricFactory;
+import org.apache.james.rrt.cassandra.CassandraMappingsSourcesDAO;
+import org.apache.james.rrt.cassandra.migration.MappingsSourcesMigration;
+import org.apache.james.task.MemoryTaskManager;
+import org.apache.james.task.Task;
+import org.apache.james.webadmin.WebAdminServer;
+import org.apache.james.webadmin.WebAdminUtils;
+import org.apache.james.webadmin.service.CassandraMappingsService;
+import org.apache.james.webadmin.service.CassandraMappingsSolveInconsistenciesTask;
+import org.apache.james.webadmin.utils.ErrorResponder;
+import org.apache.james.webadmin.utils.JsonTransformer;
+import org.eclipse.jetty.http.HttpStatus;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.restassured.RestAssured;
+import reactor.core.publisher.Mono;
+
+public class CassandraMappingsRoutesTest {
+
+    private WebAdminServer webAdminServer;
+    private MappingsSourcesMigration mappingsSourcesMigration;
+    private CassandraMappingsSourcesDAO cassandraMappingsSourcesDAO;
+    private MemoryTaskManager taskManager;
+
+    private static final String MAPPINGS_ACTION = "solveInconsistencies";
+
+    @Before
+    public void setUp() throws Exception {
+        cassandraMappingsSourcesDAO = mock(CassandraMappingsSourcesDAO.class);
```

Comment:
Can't we create the real DAO?

We can set up a Cassandra host if required.

This would enable more interesting tests like : 

```
Create s1,m1 & s2,m2 in cassandraMappingDAO
Create only s1,m1 in sourcesDAO
Run the task
sourcesDAO should contain s1,m1 & s2,m2 

Create s1,m1 & s2,m2 in cassandraMappingDAO
Create only s1,m1 & s2,m2 in sourcesDAO
Run the task
sourcesDAO should contain s1,m1
```

Which is the reason why we want to solve inconsistencies